Realm/RLMAppCredentials.h

```diff
@@ -68,6 +77,21 @@ extern RLMIdentityProvider const RLMIdentityProviderAnonymous;
  */
 + (instancetype)credentialsWithAppleToken:(RLMAppCredentialsToken)token;
 
+/**
+ Construct and return credentials for a stitch function using a document as json.
+*/
++ (instancetype)credentialsWithFunction:(NSDictionary *)payload;
```

Comment:
This method should take an error out parameter and set that and return nil rather than throwing on failure.

The name of this method indicates that the parameter is a stitch function, but it's actually the payload passed to the stitch function. Maybe `credentialsWithFunctionPayload:` or something like that?

Reply:
I've added an out param, for the swift impl I've made it as a `try` as opposed to having the user pass an `inout` Error